app/test/unit/git/stash-test.ts

```diff
@@ -46,11 +47,39 @@ describe('git/stash', () => {
       expect(stashEntries[0].branchName).toBe('master')
     })
   })
+
+  describe('createStashEntry', () => {
+    let repository: Repository
+    let readme: string
+
+    beforeEach(async () => {
+      repository = await setupEmptyRepository()
+      readme = path.join(repository.path, 'README.md')
+      await FSE.writeFile(readme, '')
+      await GitProcess.exec(['add', 'README.md'], repository.path)
+      await GitProcess.exec(['commit', '-m', 'initial commit'], repository.path)
+    })
+
+    it('creates a stash entry', async () => {
```

Comment:
I can add additional tests to cover scenarios like being in the middle of a rebase or merge, but I'm not sure of the value since we do a pretty good job of not allowing actions to take place on invalid repos.

Reply:
I'd try it out purely from a "what happens" scenario given the app might not always prevent the user from doing things - for example #6429 lets users try and merge when already in a merge, and it fails with some helpful context - and it'd be good to ensure your code behaves in a known & predictable way.

Try it out using a repository in the middle of a merge conflict (we should have scaffolding for this) - you might find yourself saying "uh-oh, we shouldn't allow any stashing at this time because XYZ" which would influence what your functions do as pre-conditions to doing stash-related work.